kirin/tasks.py

```diff
@@ -96,95 +98,66 @@ def poller(self):
+            if not locked:
+                logger.warning(
+                    "operation '{}' for contributor '{}' is already in progress".format(
+                        process_name, contributor.id
+                    )
+                )
+                return {}
 
-@celery.task(bind=True)
-def purge_trip_update_by_connector_type(self, connector_type):
-    """
-    This task will remove ONLY TripUpdate, StopTimeUpdate and VehicleJourney that are created by given connector_type
-    but the RealTimeUpdate are kept so that we can replay it for debug purpose. RealTimeUpdate will be remove by
-    another task
-    """
-    for contributor in Contributor.find_by_connector_type(connector_type.value, include_deactivated=True):
-        config = {"contributor": contributor.id, "nb_days_to_keep": contributor.nb_days_to_keep_trip_update}
-        purge_trip_update.delay(config)
-
+            logger.info("executing '{}' for contributor '{}'".format(process_name, contributor.id))
 
-@celery.task(bind=True)
-def purge_rt_update_by_connector_type(self, connector_type):
-    """
-    This task will remove realtime update
-    """
-    for contributor in Contributor.find_by_connector_type(connector_type.value, include_deactivated=True):
-        config = {"contributor": contributor.id, "nb_days_to_keep": contributor.nb_days_to_keep_rt_update}
-        purge_rt_update.delay(config)
+            deleted = self.delete_rows(contributor)
+            logger.info("operation '{}' for contributor '{}' is finished".format(process_name, contributor.id))
+            return deleted
 
+    @abstractmethod
+    def delete_rows(self, contributor: Contributor) -> Dict:
+        raise NotImplementedError
 
-@celery.task(bind=True)
-def purge_gtfs_trip_update(self):
-    purge_trip_update_by_connector_type(ConnectorType.gtfs_rt)
 
+class RealTimeUpdatePurge(AbstractPurge):
+    def delete_rows(self, contributor: Contributor) -> Dict:
+        until = datetime.date.today() - datetime.timedelta(days=int(contributor.nb_days_to_keep_rt_update))
+        return RealTimeUpdate.remove_by_contributors_until(contributors=[contributor.id], until=until)
 
-@celery.task(bind=True)
-def purge_gtfs_rt_update(self):
-    purge_rt_update_by_connector_type(ConnectorType.gtfs_rt)
 
+class TripUpdatePurge(AbstractPurge):
+    def delete_rows(self, contributor: Contributor) -> Dict:
+        until = datetime.date.today() - datetime.timedelta(days=int(contributor.nb_days_to_keep_trip_update))
+        return TripUpdate.remove_by_contributors_until(contributors=[contributor.id], until=until)
 
-@celery.task(bind=True)
-def purge_piv_trip_update(self):
-    purge_trip_update_by_connector_type(ConnectorType.piv)
 
-
-@celery.task(bind=True)
-def purge_piv_rt_update(self):
-    purge_rt_update_by_connector_type(ConnectorType.piv)
-
-
-@celery.task(bind=True)
-def purge_cots_trip_update(self):
-    purge_trip_update_by_connector_type(ConnectorType.cots)
+def _merge_deleted(original, append):
```

Comment:
The current name is related to how it's used, not what it does:
```suggestion
def _accumulate(original, append):
```
Also not fan of side-effect, but it's better for perf (except if we are clear that we consume params, but python being python...

Reply:
I'll rename it `_accumulate()` for now (see 8e7877d), but it was never a part of the code I was particularly satisfied with. All ideas are welcomed!